### DISCUSSION TITLE
Preference details not getting updated

### DISCUSSION THREAD

**Discussion by nidhikachhawa24** [Sunbird-Ed/Community#108] [Bugs] [answered]
While updating the state preferences from mobile app, the update gets failed with message: "**Profile update failed, please try again later**"

**_Tried to test -_**
**1. In Portal:** 
The state dropdown is showing only 2 states, i.e Andra Pradesh and Karnataka. The profile is getting updated.

**2. In Mobile APP:**
- _Tested in APEX APP:_ The state dropdown is showing only 2 states, i.e Andra Pradesh and Karnataka. Profile is getting Updated.
- _Tested in 4.10 version APK_: The state dropdown is showing all the states. On selecting the required values, the profile update gets failed.

cc: @swayangjit @baljeetsingh

**Comment by swayangjit** [comment, marked as answer]
@nidhikachhawa24 Can you please attach the request that is sent to the API?

**Reply by nidhikachhawa24** [reply]
@swayangjit 

**Portal**:
**Request URL:** https://dhruva.shikshalokam.org/learner/user/v3/update
**Response:**
params: {}
request: {userId: "b19c4398-1ab0-4f27-84e7-61e55621b7c9",…}
firstName: "Komal"
profileLocation: [{identifier: "0e576f3e-ee8b-49b0-9ce3-a36a4b6c0a7d", code: "28", name: "Andhra Pradesh",…},…]
0: {identifier: "0e576f3e-ee8b-49b0-9ce3-a36a4b6c0a7d", code: "28", name: "Andhra Pradesh",…}
code: "28"
id: "0e576f3e-ee8b-49b0-9ce3-a36a4b6c0a7d"
identifier: "0e576f3e-ee8b-49b0-9ce3-a36a4b6c0a7d"
name: "Andhra Pradesh"
type: "state"
1: {identifier: "6e7e8200-6ed4-4185-8003-c5955539984e", code: "2822", name: "Anantapur",…}
code: "2822"
id: "6e7e8200-6ed4-4185-8003-c5955539984e"
identifier: "6e7e8200-6ed4-4185-8003-c5955539984e"
name: "Anantapur"
parentId: "0e576f3e-ee8b-49b0-9ce3-a36a4b6c0a7d"
type: "district"
profileUserTypes: [{type: "administrator", subType: "hm"}]
0: {type: "administrator", subType: "hm"}
subType: "hm"
type: "administrator"
userId: "b19c4398-1ab0-4f27-84e7-61e55621b7c9"

![image](https://user-images.githubusercontent.com/107101132/186884665-07ba1fcd-21d2-4eca-8d06-0ecfcc7ecc57.png)

**Reply by swayangjit** [reply]
@nidhikachhawa24  I wanted the request from the mobile app where it's failing  and what is the response that you are getting.

**Reply by swayangjit** [reply]
@nidhikachhawa24  Did you resolve the issue ?

**Reply by nidhikachhawa24** [reply]
@swayangjit Yes I was able to resolve the issue. Thank You so much.
